src/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -88,28 +101,86 @@ protected Object decode(
         dateBuilder.setDateReverse(parser.nextInt(0), parser.nextInt(0), parser.nextInt(0));
         position.setTime(dateBuilder.getDate());
 
-        String type = parser.next();
+        String eventCode = parser.next();
+        String decodedAlarm = decodeAlarm(eventCode);
+        if (decodedAlarm != null) {
+            position.set(Position.KEY_ALARM, decodeAlarm(eventCode));
+        }
+        position.set(Position.KEY_EVENT, eventCode);
+
+        double batteryVoltage = parser.nextDouble();
+        if (parser.next().isEmpty()) {
+            parser.next();
+            batteryVoltage *= 0.001;
+        } else {
+            batteryVoltage += parser.nextDouble() * 0.001;
+        }
+        position.set(Position.KEY_BATTERY, batteryVoltage);
+
+        position.set(Position.KEY_TOTAL_DISTANCE, parser.nextDouble());
```

Comment:
Do not use `KEY_TOTAL_DISTANCE`. It's reserved for Traccar internal use. You should be using `KEY_ODOMETER`.